librz/core/cmd/cmd_analysis.c

```diff
@@ -8905,3 +8542,268 @@ RZ_IPI RzCmdStatus rz_rzil_vm_status_handler(RzCore *core, int argc, const char
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_hint_set_optype_handler(RzCore *core, int argc, const char **argv) {
+	int type = rz_analysis_optype_from_string(argv[1]);
+	if (type < 0) {
+		return RZ_CMD_STATUS_WRONG_ARGS;
+	}
+	rz_analysis_hint_set_type(core->analysis, core->offset, type);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_hint_del_optype_handler(RzCore *core, int argc, const char **argv) {
+	rz_analysis_hint_unset_type(core->analysis, core->offset);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_hint_set_immbase_handler(RzCore *core, int argc, const char **argv) {
+	int base = rz_num_base_of_string(core->num, argv[1]);
+	if (argc == 3) {
+		ut64 nword = rz_num_math(core->num, argv[2]);
+		rz_analysis_hint_set_nword(core->analysis, core->offset, (int)(nword));
+	}
+	rz_analysis_hint_set_immbase(core->analysis, core->offset, base);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_hint_del_immbase_handler(RzCore *core, int argc, const char **argv) {
+	rz_analysis_hint_unset_immbase(core->analysis, core->offset);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_hint_set_offset_handler(RzCore *core, int argc, const char **argv) {
+	RzAnalysisOp op = { 0 };
+	ut8 code[128] = { 0 };
+	if (!rz_io_read_at(core->io, core->offset, code, sizeof(code))) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzCmdStatus res = RZ_CMD_STATUS_ERROR;
+	int ret = rz_analysis_op(core->analysis, &op, core->offset, code, sizeof(code), RZ_ANALYSIS_OP_MASK_VAL);
+	if (ret < 1) {
+		goto exit;
+	}
+	// HACK: Just convert only the first imm seen
+	ut64 offimm = 0;
+	for (int i = 0; i < 3; i++) {
+		if (op.src[i]) {
+			if (op.src[i]->imm) {
+				offimm = op.src[i]->imm;
+			} else if (op.src[i]->delta) {
+				offimm = op.src[i]->delta;
+			}
+		}
+	}
+	if (!offimm && op.dst) {
+		if (op.dst->imm) {
+			offimm = op.dst->imm;
+		} else if (op.dst->delta) {
+			offimm = op.dst->delta;
+		}
+	}
+	if (!offimm) {
+		goto exit;
+	}
+	// TODO: Allow to select from multiple choices
+	RzList *otypes = rz_type_db_get_by_offset(core->analysis->typedb, offimm);
+	RzListIter *iter;
+	RzTypePath *tpath;
+	rz_list_foreach (otypes, iter, tpath) {
+		// TODO: Support also arrays and pointers
+		if (tpath->typ->kind == RZ_TYPE_KIND_IDENTIFIER) {
+			if (!strcmp(argv[1], tpath->path)) {
+				rz_analysis_hint_set_offset(core->analysis, core->offset, tpath->path);
+				break;
+			}
+		}
+	}
+	rz_list_free(otypes);
+	res = RZ_CMD_STATUS_OK;
+exit:
+	rz_analysis_op_fini(&op);
+	return res;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_hint_del_offset_handler(RzCore *core, int argc, const char **argv) {
+	rz_analysis_hint_unset_offset(core->analysis, core->offset);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_list_struct_offsets_handler(RzCore *core, int argc, const char **argv) {
+	ut64 toff = rz_num_math(core->num, argv[1]);
+	if (!toff) {
+		return RZ_CMD_STATUS_WRONG_ARGS;
+	}
```

Comment:
I think this should be replaced with `rz_num_is_valid_input`

Reply:
1) About check on zero: zero offset is not allowed.
2) About `rz_num_is_valid_input`: I expect that rzshell validates arguments of type `RZ_CMD_ARG_TYPE_NUM`/`RZ_CMD_ARG_TYPE_RZNUM`

Reply:
> About check on zero: zero offset is not allowed

Ah ok

> About rz_num_is_valid_input: I expect that rzshell validates arguments of type RZ_CMD_ARG_TYPE_NUM/RZ_CMD_ARG_TYPE_RZNUM

This is a good point actually. It is not done at the moment, but maybe it should be done indeed.

Reply:
Mmmm why zero offset is not allowed by the way?

Reply:
Because all structs have field at zero offset.